README_DOCKER.md

```diff
@@ -0,0 +1,223 @@
+| with_mysql     | WITH_MYSQL            | true          |
+| with_sqlite3   | WITH_SQLITE3          | false         |
+| with_pgsql     | WITH_PGSQL            | false         |
+| with_odbc      | WITH_ODBC             | false         |
+
+And some extra build args which is not related to CMAKE:
+
+| Build Argument | Description                       | Default Value |
+|----------------|-----------------------------------|---------------|
+| git_repo       | The repository you want to clone  | this repo url |
+| git_branch     | The branch you want to build from | develop       |
+
+Adding above build arguments to enable/disable features (Image tagging will helps you distinguish image capabilities/features)
+
+There are some enviroment variables you might want to override:
+
+| Env Var     | Description                                             | Default Value |
+|-------------|---------------------------------------------------------|---------------|
+| SERVER_TYPE | Server type, choose between `d2cs`, `d2dbs` and `bnetd` | `bnetd`       |
+
+Choose the appropriate `SERVER_TYPE` base on your image build
+
+Below are some sample build commands:
+
+```bash
+# This command will use all default build arguments and build PVPGN docker image with only bnetd and mysql support using master branch
+docker build . -t pvpgn-server:bnetd-mysql
+
+# This command will build a PVPGN with only d2cs and sqlite3 support using develop branch
+docker build . \
+  --build-arg with_d2cs=true \     # enable d2cs support (disable by default)
+  --build-arg with_bnetd=false \   # enable bnetd support (enable by default)
+  --build-arg with_mysql=false \   # enable mysql support (enable by default)
+  --build-arg with_sqlite3=true \  # enable sqlite3 support (disable by default)
+  --build-arg git_branch=develop \ # switch to develop branch (master branch by default)
+  -t pvpgn-server:d2cs-sqlite      # tag the image 
+```
+
+## How to run and manage:
+
+### Using docker only:
+
+To run the images, you will need to mount existed configuration files and assets which is located at `/usr/local/etc/pvpgn` and `/usr/local/var/pvpgn` inside the container. The example bellow shows how to obtain configuration files and turn up some pvpgn servers with `mysql` support:
+
+1. Prepare images:
+
+```bash
+docker build . -t pvpgn-server:bnetd-mysql
+docker build . --build-arg with_d2cs=true --build-arg with_bnetd=false -t pvpgn-server:d2cs-mysql
+docker build . --build-arg with_d2dbs=true --build-arg with_bnetd=false -t pvpgn-server:d2dbs-mysql
+```
+
+2. Prepare configuration files and assets (skip this step if you've already had them):
+
+```bash
+# create configuration and asset locations
+mkdir -p /your/config/dir /your/assets/dir
+# copy configuration files and base assets from an image, all base configuration files in any image are the same
+docker run -v /your/config/dir:/tmp/conf --rm --entrypoint cp pvpgn-server:bnetd-mysql -r /usr/local/etc/pvpgn/* /tmp/conf
```

Comment:
> 
> 
> Pretty sure the image name should be at the end otherwise it is grabbed by cp command. I get:
> 
> sudo docker run -v /opt/pvpgn/etc:/tmp/conf --rm --entrypoint cp pvpgn-server:bnetd-mysql -r /usr/local/etc/pvpgn/* /tmp/conf
> cp: cannot stat '/usr/local/etc/pvpgn/bnetd_default_user.cdb': No such file or directory

I've already overwrite the entrypoint of the container to `cp`. Normally it will be `/bin/sh`. If you run a docker command like this:
```sh
docker run some-image some-command
```
It means that you are running `/bin/sh some-command`. But changing the entrypoint to `cp`, running a command like this:
```sh
docker run --entrypoint cp some-image -r /dir1/* /dir2
```
the container will execute your command like this: `cp -r /dir1/* /dir2`. That's why you can run `mysql` image like this:
```sh
docker run mysql --default-authentication-plugin=mysql_native_password
```

Actually the error is caused by `cp` command somehow couldn't recognize the wildcard directory and got it as a directory or file. I don't know why this problem happens. Fixing it is pretty easy, just change the `-r /usr/local/etc/pvpgn/* /tmp/conf` to `-a /usr/local/etc/pvpgn/. /tmp/conf`.